### PR TITLE
[loss/unittest] Proper support for loss as layer

### PR DESCRIPTION
**Pull request by kparichay**
Support for loss as a layer with the API has been done.
However, the verification of this feature was missing.
This patch ensures that the added loss layer is verified and its
required conditions checked before finalizing the graph.

Add loss layer unittests for various loss configurations to be
supported. These tests do not ensure correct graph formation.
This will be done soon with models unittest.

Add golden tests for loss layer added as a layer.
    This patch modified the existing models test architecture to be
    described where layer is added externally.
    
More will be added ones more models tests are enabled.

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>
cc. @zhoonit

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/graph/network_graph.cpp [hunk 1]
```diff
@@ -221,72 +229,76 @@ int NetworkGraph::realizeMultiOutputType(
...
-      status = ML_ERROR_INVALID_PARAMETER;
-    return status;
-  }
-
-  if (loss_type.empty()) {
-    return status;
+void NetworkGraph::finalizeLossLayer() {
+  auto loss_layer_node = getSortedLayerNode(graph.size() - 1);
+  std::string loss_type = loss_layer_node->getType();
+  if (graph.size() == 1) {
```
Comment (zhoonit) [inline nntrainer/graph/network_graph.cpp:235]: Could you explain what this line is checking?

is it checking that graph is only consists of a loss?

**Review by zhoonit — APPROVED** [review]
Thanks :)

File: nntrainer/graph/network_graph.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/graph/network_graph.cpp:235]: For the naive cross-entropy layer, the previous layer must be softmax or sigmoid. So, if graph size is 1, then its not possible. However, other losses/layers are possible.
This also provides added security that the check at L245 does not fail.
